qiskit/pulse/instructions/play.py

```diff
@@ -14,64 +14,104 @@
-        """Create a new pulse instruction.
+    def __init__(
+        self,
+        pulse: Pulse,
+        *,
+        target: Optional[PulseTarget] = None,
+        frame: Optional[Frame] = None,
+        mixed_frame: Optional[MixedFrame] = None,
+        channel: Optional[PulseChannel] = None,
+        name: Optional[str] = None,
+    ):
+        """Create a new pulse play instruction.
+
+        To uniquely identify the mixed frame to which the pulse is applied, the arguments must include
+        exactly one of ``mixed_frame``, ``channel`` or the duo ``target`` and ``frame``.
 
         Args:
             pulse: A pulse waveform description, such as
                    :py:class:`~qiskit.pulse.library.Waveform`.
+            target: The target to which the pulse is applied (must be provided with ``frame``).
+            frame: The frame characterizing the pulse carrier (must be provided with ``target``).
+            mixed_frame: The mixed frame to which the pulse is applied.
             channel: The channel to which the pulse is applied.
             name: Name of the instruction for display purposes. Defaults to ``pulse.name``.
-        """
-        if name is None:
-            name = pulse.name
-        super().__init__(operands=(pulse, channel), name=name)
-
-    def _validate(self):
-        """Called after initialization to validate instruction data.
 
         Raises:
+            PulseError: If the combination of ``target``, ``frame``, ``mixed_frame`` and ``channel``
+                doesn't specify a unique mixed frame.
+            PulseError: If the inputs to ``target``, ``frame``, ``mixed_frame`` and ``channel``
+                are not of the appropriate type.
             PulseError: If pulse is not a Pulse type.
-            PulseError: If the input ``channel`` is not type :class:`PulseChannel`.
         """
-        if not isinstance(self.pulse, Pulse):
+        if not isinstance(pulse, Pulse):
             raise PulseError("The `pulse` argument to `Play` must be of type `library.Pulse`.")
 
-        if not isinstance(self.channel, PulseChannel):
-            raise PulseError(f"Expected a pulse channel, got {self.channel} instead.")
+        if (target is not None) != (frame is not None):
+            raise PulseError("target and frame must be provided simultaneously")
+        if (channel is not None) + (mixed_frame is not None) + (target is not None) != 1:
+            raise PulseError(
+                "Exactly one of mixed_frame, channel or the duo target and frame has to be provided"
+            )
+
+        if target is not None:
+            if not isinstance(target, PulseTarget):
+                raise PulseError(f"Expected a PulseTarget, got {target} instead.")
+            if not isinstance(frame, Frame):
+                raise PulseError(f"Expected a Frame, got {frame} instead.")
+            mixed_frame = MixedFrame(target, frame)
+        else:
+            mixed_frame = mixed_frame or channel
```

Comment:
```suggestion
        if (channel is not None) + (mixed_frame is not None) + (target is not None) != 1:
            raise PulseError(
                "Exactly one of mixed_frame, channel or the duo target and frame has to be provided"
            )

        if target is not None:
            if not isinstance(target, PulseTarget):
                raise PulseError(f"Expected a PulseTarget, got {target} instead.")
            if not isinstance(frame, Frame):
                raise PulseError(f"Expected a Frame, got {frame} instead.")
            mixed_frame = MixedFrame(target, frame)
        else:
            mixed_frame = mixed_frame or channel
```
Also equivalent right?

Reply:
This will actually fail to raise an error if you provide `frame` and `mixed_frame` (the frame argument will be ignored).